storage/dataflux/sequential.go

```diff
@@ -33,37 +33,37 @@ const (
 // If the next token is empty, then listing is complete.
 func (c *Lister) sequentialListing(ctx context.Context) ([]*storage.ObjectAttrs, string, error) {
 	var result []*storage.ObjectAttrs
-	var objectsListed int
+	var objectsIterated int
 	var lastToken string
 	objectIterator := c.bucket.Objects(ctx, &c.query)
 	objectIterator.PageInfo().Token = c.pageToken
 	objectIterator.PageInfo().MaxSize = defaultPageSize
 
 	for {
-		objects, nextToken, numObjects, err := doSeqListing(objectIterator, c.skipDirectoryObjects)
+		objects, nextToken, pageSize, err := doSeqListing(objectIterator, c.skipDirectoryObjects)
 		if err != nil {
 			return nil, "", fmt.Errorf("failed while listing objects: %w", err)
 		}
 		result = append(result, objects...)
 		lastToken = nextToken
-		objectsListed += numObjects
-		if nextToken == "" || (c.batchSize > 0 && objectsListed >= c.batchSize) {
+		objectsIterated += pageSize
+		if nextToken == "" || (c.batchSize > 0 && objectsIterated >= c.batchSize) {
 			break
 		}
 		c.pageToken = nextToken
 	}
 	return result, lastToken, nil
 }
 
-func doSeqListing(objectIterator *storage.ObjectIterator, skipDirectoryObjects bool) (result []*storage.ObjectAttrs, token string, objectsListed int, err error) {
+func doSeqListing(objectIterator *storage.ObjectIterator, skipDirectoryObjects bool) (result []*storage.ObjectAttrs, token string, pageSize int, err error) {
 
 	for {
 		attrs, errObjectIterator := objectIterator.Next()
-		objectsListed++
 		// Stop listing when all the requested objects have been listed.
 		if errObjectIterator == iterator.Done {
 			break
 		}
+		pageSize++
```

Comment:
Nice catch! Shouldn't this go even later, after the err.. != nil check?

Reply:
Done!